shell/common/rasterizer.cc

```diff
@@ -137,6 +148,10 @@ void Rasterizer::NotifyLowMemoryWarning() const {
         << "Rasterizer::NotifyLowMemoryWarning called with no GrContext.";
     return;
   }
+  auto context_switch = surface_->MakeRenderContextCurrent();
+  if (!context_switch->GetResult()) {
+    return;
+  }
```

Comment:
I think this might be wrong.

This means that if the OS sends a low memory warning, we will fail to clear the cache if we're unable to make the context current. This will result in the OS killing us or some other process(es), particularly on mobile/Fuchsia/CastOS. 

If we're running low on memory, we should unconditionally purge the resource cache. This will not result in context corruption, though it may slow down next frames.

Reply:
Filed https://github.com/flutter/flutter/issues/105192 to discuss